hypershift-operator/controllers/nodepool/azure.go

```diff
@@ -55,11 +62,23 @@ func generateSSHPubkey() (string, error) {
 	return base64.StdEncoding.EncodeToString(ssh.MarshalAuthorizedKey(publicRsaKey)), nil
 }
 
-func bootImage(hcluster *hyperv1.HostedCluster, nodepool *hyperv1.NodePool) string {
-	if nodepool.Spec.Platform.Azure.ImageID != "" {
-		return nodepool.Spec.Platform.Azure.ImageID
+func bootImage(hcluster *hyperv1.HostedCluster, nodepool *hyperv1.NodePool) (string, error) {
+	var bootImageGalleyID string
+	splitResourceGroupName := strings.Split(hcluster.Spec.Platform.Azure.ResourceGroupName, "-")
+	if len(splitResourceGroupName) <= 0 {
+		return "", fmt.Errorf("failed to parse resource group name unique ID to retrieve boot image: %s", hcluster.Spec.Platform.Azure.ResourceGroupName)
+	} else {
+		bootImageGalleyID = splitResourceGroupName[len(splitResourceGroupName)-1]
+	}
+
+	switch nodepool.Spec.Arch {
+	case "amd64":
+		return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/BootImageGallery_%s/images/RHCOS_Amd64/versions/1.0.0", hcluster.Spec.Platform.Azure.SubscriptionID, hcluster.Spec.Platform.Azure.ResourceGroupName, bootImageGalleyID), nil
```

Comment:
this can be a parametarised function so you don't need to handcraft it again in the tests

Reply:
Yeah that is a good idea. The function has been updated.